### DISCUSSION TITLE
my custom github pages deploy action seemed to work but getting 404 when trying to access page

### DISCUSSION THREAD

**Discussion by jamiepratt** [community/community#29250] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi,

I'm trying to do a custom build of some static pages to serve through Github pages.

[The workflow](https://github.com/jamiepratt/jointprob-clerk/blob/master/.github/workflows/main.yml) runs without errors and the deployment seems to work:

<img width="813" alt="Screenshot 2022-08-12 at 10 41 41" src="https://user-images.githubusercontent.com/558780/184321473-cdad1956-2348-4c81-8023-118fb7d3d111.png">

And I see a gh-pages branch has been created and [an index.html](https://github.com/jamiepratt/jointprob-clerk/blob/gh-pages/index.html) has been added to it.

But I can't find the URL to access my pages. I've tried:

https://jamiepratt.github.io/jointprob-clerk/index.html
https://jamiepratt.github.io/jointprob-clerk/
https://jamiepratt.github.io/index.html

and

https://jamiepratt.github.io/

All 404s.

Hoping someone can help.

I don't see somewhere to tell github where to find my pages if I'm doing a custom deploy, I assume that gh should find the pages automatically.

Thanks for any help,

Jamie

**Comment by manasesjesus** [comment]
I checked your repo and it seems Pages is not activated. 
Go to Settings > Pages, and choose if you want to deploy from a branch or using Actions (this last one is still in beta).

<img width="993" alt="zgh" src="https://user-images.githubusercontent.com/24204142/184334419-f843e316-b7e6-42b9-b8c9-815a90a03f91.png">

**Comment by jamiepratt** [comment]
Hi,

Thanks for your help!

What do you mean Pages is not activated? I have selected "GitHub Actions":

----------------------------------------------------------------------------------------------------------------

<img width="735" alt="Screenshot 2022-08-12 at 15 56 08" src="https://user-images.githubusercontent.com/558780/184369432-8f9a6f54-f2ba-4b89-a132-19283b207aca.png">
----------------------------------------------------------------------------------------------------------------

And I have [a workflow file](https://github.com/jamiepratt/jointprob-clerk/blob/master/.github/workflows/main.yml) in the correct place in my repo.

Oddly I do not see any details of workflow runs here:

----------------------------------------------------------------------------------------------------------------
<img width="780" alt="Screenshot 2022-08-12 at 16 01 04" src="https://user-images.githubusercontent.com/558780/184369559-f57d9257-19c2-432d-90e5-0175b3ca4ce2.png">
----------------------------------------------------------------------------------------------------------------

But if I click the link "View workflow runs", I do see the workflow has been completed successfully as you can see in the screenshot I posted earlier.

Jamie

**Comment by jamiepratt** [comment]
I've got it working by switching to the 'deploy from branch' source option. The files are there in the gh-pages branch.

But this is just a workaround. I want to use Github Actions. The build process is being triggered by pushing to the master branch and files are being put in the gh-pages branch but for some reason, the updated files are not being served by github.io, only the files where I have forced an update by switching back to 'deploy from branch'.

**Reply by chrispat** [reply]
If you are pushing the built content back to a branch in the repository then you will need to use the "Deploy from branch" model in order to see your new content.  If you want to use Actions to deploy then you need to use the new deploy pages actions.  There are a number of example workflows https://github.com/actions/starter-workflows/tree/main/pages that show up in the new workflow screen in Actions.

**Comment by manasesjesus** [comment]
It seems your workflow file doesn't have all the info. Check the suggested workflows to implement it. In your case, you'd need the [static template](https://github.com/actions/starter-workflows/blob/main/pages/static.yml).

And it will show the details once its done.

<img width="785" alt="zg" src="https://user-images.githubusercontent.com/24204142/184385140-f056684b-edfb-4f1c-bef3-54b799836f89.png">

**Comment by jamiepratt** [comment, marked as answer]
This is what I did in the end:

https://github.com/jamiepratt/jointprob-clerk/blob/master/.github/workflows/main.yml

I based my workflow on another:

https://github.com/actions/starter-workflows/blob/main/pages/jekyll-gh-pages.yml

And just replaced the jekyll build with the clojure and clerk build process.

The JamesIves/github-pages-deploy-action didn't work for me.

**Comment by Matthommes** [comment]
<img width="1680" alt="Screenshot 2023-08-31 at 6 43 17 pm" src="https://github.com/community/community/assets/53288286/735dba49-41e6-4812-abdb-06472e49b2c9">
Hello, Mine doesn't show the gh-pages when I try to re-run job
